fix(bedrock, mostly): Update terminology (#545)
ethereum-optimism/community-hub: src/docs/developers/bedrock.md

### Withdrawals (from Optimism to Ethereum)

The withdrawal process is similar, but not identical.

1. You initialize withdrawals with an L2 transaction, the same you did prior to bedrock.

> 1. After the new state root is submitted to L1 (up to an hour on mainnet, less than that on the test network) submit the withdrawal proof using `proveMessage`.

Reviewer: Slightly different wording to:
1. Clarify that the action is not up to the user
2. output root not state root (do we need to change that in more places?)
3. fix the function name.

```suggestion
1. Wait for the next output root to be submitted to L1 (up to an hour on mainnet, less than that on the test network) then submit the withdrawal proof using `proveWithdrawalTransaction`.
```

The information in this section is primarily useful to people who run an Optimism network node, either as a replica or as an independent development node.

In bedrock processing is divided between two components:

- **op-node**: 
  The component responsible for deriving the L2 chain from L1 blocks.

>   It is somewhat similar to the consensus layer client.

Reviewer 2: Add a lil explainer as if I don't know what a consensus client is.

> - **op-geth**: 

Reviewer 2: Again high level explainer of what this is.
